behind-the-veil-siteroot/imports/ui/components/pages/add-availability/AddAvailability.jsx

```diff
@@ -0,0 +1,276 @@
+            "artistUsername": "username"
+        }
+    ]
+
+    const initialAvailability = MOCK_AVAILABILITY.find(entry => entry.artistUsername === artistUsername) || {
+        _id: useId(),
+        availabilityTimes: {},
+        artistUsername: artistUsername
+    };
+
+    // form input values
+    const [inputs, setInputs] = useState({
+        date: startOfDay(new Date()),
+        times: [],
+    });
+
+    const [availability, setAvailability] = useState(
+        initialAvailability.availabilityTimes
+    );
+
+    // id's
+    const dateInputId = useId();
+    const timeInputId = useId();
+
+    /**
+     * Calculate available times that the user can select, based on date
+     * @param {Date} date day at which we want the available time slots
+     * @returns array of date objects that correspond to available times, on the hour
+     */
+    const getAvailableTimes = (date) => {
+        if (!(isValid(date) && isDate(date))) {
+            console.warn('invalid date')
+            return []
+        }
+
+        const hours = eachHourOfInterval({
+            start: set(date, {hours: 6, minutes: 0, seconds: 0}),
+            end: set(date, {hours: 19, minutes: 0, seconds: 0})
+        })
+
+        return hours
+    };
+
+    const handleSave = (event) => {
+        event.preventDefault();
+
+        const dateKey = format(inputs.date, "yyyy-MM-dd");
+
+        if (!initialAvailability.availabilityTimes[dateKey]) {
+            initialAvailability.availabilityTimes[dateKey] = [];
+        }
+
+        inputs.times.forEach((time) => {
+            const hour = time.getHours();
+            if (!initialAvailability.availabilityTimes[dateKey].includes(hour)) {
+                initialAvailability.availabilityTimes[dateKey].push(hour);
+            }
+        });
+
+        // TODO: add/update availability in database
+        const existingIndex = MOCK_AVAILABILITY.findIndex(entry => entry.artistUsername === initialAvailability.artistUsername);
+        if (existingIndex !== -1) {
+            MOCK_AVAILABILITY[existingIndex] = initialAvailability;
+        } else {
+            MOCK_AVAILABILITY.push(initialAvailability);
+        }
+    }
+
+    const handleManualDateInput = (event) => {
+        const dateInput = event.target.value;
+
+        // if manual input is a valid date within the allowable interval
+        // convert it to a date object and store it
+        const dateFormat = /^(0[1-9]|[12][0-9]|3[01])-(0[1-9]|1[0-2])-(19|20)\d{2}$/
+        if (dateFormat.test(dateInput)) {
+            const parsedDate = parse(dateInput, 'dd-MM-yyyy', new Date())
+            if (
+                isValid(parsedDate) &&
+                isWithinInterval(parsedDate, VALID_INTERVAL)
+            ) {
+                setInputs((i) => ({...i, date: parsedDate}));
+                return
+            }
+        }
+
+        // else, just update date input with the raw value
+        //setInputs((i) => ({...i, date: dateInput}));
```

Comment:
Is this supposed to be commented out? Without this line, the manual date input logic doesn't work

Reply:
It was not supposed to, I have updated it